rfcs/text/0002_encrypted_attributes.md

```diff
@@ -0,0 +1,228 @@
+from the request using public `spaces` plugin API.
+
+## Encryption and decryption
+
+Saved object attributes are encrypted using [@elastic/node-crypto](https://github.com/elastic/node-crypto) library. Please
+take a look at the source code of this library to know how encryption is performed exactly, what algorithm and encryption 
+parameters are used, but in short it's AES Encryption with AES-256-GCM that uses random initialization vector and salt.
+
+As with encryption key for Kibana's session cookie, master encryption key used by `encrypted_saved_objects` plugin can be
+defined as a configuration value (`xpack.encrypted_saved_objects.encryptionKey`) via `kibana.yml`, but it's **highly 
+recommended** to define this key in the [Kibana Keystore](https://www.elastic.co/guide/en/kibana/current/secure-settings.html)
+instead. The master key should be cryptographically safe and be equal or greater than 32 bytes.
+
+To prevent certain vectors of attacks where raw content of encrypted attributes of one saved object is copied to another
+saved object that would unconsciously allow to decrypt content that was not supposed to be decrypted we rely on Additional
+authenticated data (AAD) during encryption and decryption. AAD consists of the following components:
+
+* Saved object ID
+* Saved object type
+* Saved object attributes
+ 
+AAD does not include encrypted attributes themselves and attributes defined in optional `attributesToExcludeFromAAD` 
+parameter provided during saved object type registration with `encrypted_saved_objects` plugin.
+
+## Audit
+
+Encrypted attributes will most likely contain a sensitive information and any attempt to access these should be properly
+logged to allow any further audit procedures. The following events will be logged with Kibana audit log functionality:
+
+* Successful attempt to encrypt attributes (incl. saved object ID, type and attributes names)
+* Failed attempt to encrypt attribute (incl. saved object ID, type and attribute name)
+* Successful attempt to decrypt attributes (incl. saved object ID, type and attributes names)
+* Failed attempt to decrypt attribute (incl. saved object ID, type and attribute name)
+
+In addition to audit log events we'll issue ordinary log events for any attempts to save, update or decrypt saved objects
+with missing attributes that were supposed to be encrypted/decrypted based on the registration parameters. 
+
+# Benefits
+
+* None of the registered types will expose their encrypted details. The saved
+objects with their unencrypted attributes could still be obtained and searched
+on. The wrapper will follow all the security and spaces filtering of saved
+objects so that only users with appropriate permissions will be able to obtain
+the scrubbed objects or _save_ objects with encrypted attributes.
+
+* No explicit access to a method that takes in an encrypted string exists. And no
+access to the private key used is exposed through the plugin. If the type was not
```

Comment:
This isn't strictly true as plugins can access each other's config.

Reply:
Yeah, will remove this sentence.